doc/source/text.rst

```diff
@@ -199,6 +199,143 @@ regular expression object will raise a ``ValueError``.
     ---------------------------------------------------------------------------
     ValueError: case and flags cannot be set when pat is a compiled regex
 
+.. _text.concatenate:
+
+Concatenation
+-------------
+
+There are several ways to concatenate a ``Series``, either with itself or others, all based on :meth:`~Series.str.cat`.
+
+Concatenating a single Series into a string
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The content of a ``Series`` can be concatenated:
+
+.. ipython:: python
+
+    s = pd.Series(['a', 'b', 'c', 'd'])
+    s.str.cat(sep=',')
+    
+If not specified, the keyword ``sep`` for the separator defaults to ``''``:
+
+.. ipython:: python
+
+    s.str.cat()
+
+By default, missing data is ignored. Using ``na_rep``, they can be given a representation:
+
+.. ipython:: python
+
+    t = pd.Series(['a', 'b', np.nan, 'd'])
+    t.str.cat(sep=',')
+    t.str.cat(sep=',', na_rep='-')
+
+Concatenating a Series and something one-dimensional into a Series
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first argument to :meth:`~Series.str.cat` (which corresponds to the keyword ``others``) can be used to specify values that should be concatenated correspondingly. 
+The result will be a ``Series`` of strings. Possible input types are ``list``, ``Series`` and ``np.ndarray``, all of which must match the length of
+the calling ``Series``.
+
+.. ipython:: python
+
+    s.str.cat(['A', 'B', 'C', 'D'])
+    s.str.cat(s, sep = ':')
+    s.str.cat(s.values, sep = '.')
+    
+Missing values will remain missing in the result, *unless* ``na_rep`` is specified:
+
+.. ipython:: python
+
+    s.str.cat(t)
+    s.str.cat(t, na_rep='-')
+
+Concatenating a Series and something two-dimensional into a Series
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. versionadded:: 0.23.0
+
+The parameter ``others`` can also be two-dimensional, i.e. a ``DataFrame`` or a ``np.ndarray`` with ``ndim==2``. In this case, the number or rows must match
+the lengths of the calling ``Series``.
+
+.. ipython:: python
+
+    d = pd.concat([t, s], axis=1)
+    s.str.cat(d, na_rep='-')
+    s.str.cat(d.values)
+    
+Concatenating a Series and an indexed object into a Series, with alignment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. versionadded:: 0.23.0
+
+For concatenation with a ``Series`` or ``DataFrame``, it is possible to align the respective indexes before concatenation by setting ``align=True``.
+
+.. ipython:: python
+
+    u = pd.Series(['b', 'd', 'a', 'c'], index=[1, 3, 0, 2])
+    s.str.cat(u, align=False)
+    s.str.cat(u, align=True)
+
+.. warning::
+
+    If the ``align`` keyword is not passed, the method :meth:`~Series.str.cat` will currently fall back to the behavior before version 0.23.0 (i.e. ``align=False``),
```

Comment:
if we have join, then we don't need an align keyword

Reply:
It get's a bit crowded to then still have the choice of not aligning, and ``None`` for API-compatibility. Would you be thinking about something like (removing ``join`` instead of ``align``):
``align = None|False|'left'|'inner'|'outer'|'right'``?

My personal preference would be the align/join combination.

Reply:
I guess it would also work with ``align = None|'left'|'inner'|'outer'|'right'``, if we *really* want to discourage working with unaligned data (the user could still use ``.values`` to circumvent alignment after all - like in most of pandas).